tiledb/sm/cpp_api/query.h

```diff
@@ -1086,6 +1086,79 @@ class Query {
         sizeof(char));
   }
 
+  /**
+   * Gets a buffer for a fixed-sized attribute.
+   *
+   * @param attr Attribute name
+   * @param buff Buffer array pointer with elements of the attribute type.
+   * @param nelements Number of array elements.
+   * @param element_size Size of array elements (in bytes).
+   **/
+  Query& get_buffer(
+      const std::string& attr,
+      void** data,
+      uint64_t** data_nelements,
+      uint64_t* element_size) {
+    auto ctx = ctx_.get();
+
+    auto elem_size_iter = element_sizes_.find(attr);
+    if (elem_size_iter == element_sizes_.end()) {
+      throw TileDBError(
+          "[TileDB::C++API] Error: No buffer set for attribute '" + attr +
+          "'!");
+    }
+
+    ctx.handle_error(tiledb_query_get_buffer(
+        ctx.ptr().get(), query_.get(), attr.c_str(), data, data_nelements));
+
+    **data_nelements = (**data_nelements) / elem_size_iter->second;
```

Comment:
If we expect this to always divide evenly, let's add an `assert(... % ... == 0)`. Same in the other function. I'm only concerned about this because the process could continue for a while before we notice there is a problem. I'm not as pedantic about checking that `elem_size_iter->second != 0` because a divide-by-zero will be immediately obvious.

Reply:
Done